### PR TITLE
Port to Gnuradio 3.8

### PR DESCRIPTION
**Pull request by StefanBruens**
- Use new syntax for required GR components
- Use gnuradio::gnuradio-* instead of GNURADIO_ALL_LIBRARIES
- Update sources for renamed/consolidated headers

### CODE REVIEW

**Comment by alexf91** [comment]
Porting is not so simple. GNU Radio 3.7 will likely be around for years, because many applications are poorly maintained and won't be port to 3.8. It appears to be relatively easy to support both version in Gqrx.

Here are more discussions:
Existing PR: https://github.com/csete/gqrx/pull/696
Discussion on porting and support for 3.7/3.8: https://github.com/csete/gqrx/issues/657

BTW, what is my commit doing in your PR?

**Comment by StefanBruens** [comment]
When you have a ported gr-osmosdr, it actually *is* so simple.

I have an almost complete set of OOT GR modules updated to GR 3.8, porting is not really hard after the first package. The packages I have omitted are mostly ones which also depend on e.g. Qt4, which is EOL for years.

Unfortunately, Github seems to be unable to track common commits, my branch was branched from the v2.11.5 tag.

**Comment by alexf91** [comment]
> When you have a ported gr-osmosdr, it actually is so simple.

Sure, but we should wait for distributions before switching Gqrx to 3.8-only. It makes little sense to switch to 3.8 if popular distributions like Ubuntu can't run new versions. If we only have to include different headers and have some different names, then we can support both versions.

**Comment by alexf91** [comment]
@StefanBruens I have rebased your branch: https://github.com/alexf91/gqrx/tree/gnuradio3.8

**Comment by csete** [comment]
Thanks for the patch; however, as we also discussed in issue #657, it would be nice to support both versions as long as the changes are relatively simple.

File: CMakeLists.txt [hunk 1]
```diff
@@ -99,15 +99,9 @@ endfunction(add_source_files)
 
 # 3rd Party Dependency Stuff
 find_package(Qt5 COMPONENTS Core Network Widgets Svg REQUIRED)
-find_package(Boost COMPONENTS system program_options REQUIRED)
```
Comment (csete) [inline CMakeLists.txt]: These Boost components are required by Gqrx

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -58,12 +58,17 @@ endif(WIN32)
 #######################################################################################################################
 # Build the program
 add_executable(${PROJECT_NAME} ${${PROJECT_NAME}_SOURCE} ${UIS_HDRS} ${RESOURCES_LIST})
-qt5_use_modules(${PROJECT_NAME} Core Network Widgets Svg)
```
Comment (csete) [inline src/CMakeLists.txt]: Why change this? Even is the other way is preferred, is the change related to the porting?

**Comment by csete** [comment]
I have now merged the patch from @alexf91 so gqrx should now build with both gnuradio 3.7 and 3.8.

**Pull request closed without merging**